client/assets/js/main.js

```diff
@@ -0,0 +1,343 @@
+    $('#home-page').show()
+    $('#show-todo').hide()
+    $('#no-todo-found').hide()
+    $('#newTodoForm').hide()
+    $('#todo-update').hide()
+    $('#btn-login').show()
+    $('#btn-register').show()
+    $('#dropdown-action').hide()
+    $('#all-todo').hide()
+    $('#show-todo').hide()
+}
+
+function loggedUserLandingPage() {
+    $('#home-page').show()
+    $('#show-todo').hide()
+    $('#no-todo-found').hide()
+    $('#newTodoForm').hide()
+    $('#todo-update').hide()
+    $('#btn-login').hide()
+    $('#btn-register').hide()
+    $('#show-todo').hide()
+    $('#all-todo').hide()
+    $('#dropdown-action').show()
+    $('#username-actions').empty()
+    $('#username-actions').append(`<img src="https://robohash.org/${localStorage.currentUser}?set=set5" width="23" height="23" alt="${localStorage.currentUser} icon"> ${localStorage.currentUser}`)
+}
+
+function register() {
+    let name = $('#input-new-name').val()
+    let email = $('#input-new-email').val()
+    let password = $('#input-new-password').val()
+    $.ajax({
+        method: 'POST',
+        url: 'http://localhost:3000/register',
+        data: {
+            name,
+            email,
+            password
+        }
+    })
+        .done(({data}) => {
+            console.log(data)
+            $('#registerForm').modal('hide')
+            $('.error-message').empty()
+            setTimeout(() => {
+                $('#loginForm').modal('show')
+            }, 400)
+        })
+        .fail(err => {
+            $('.error-message').empty()
+            $('.error-message').append(`<small class="form-text text-muted">${err.responseJSON.msg}</small>`)
+        })
+}
+
+function login() {
+    let email = $('#input-email').val()
+    let password = $('#input-password').val()
+    $.ajax({
+        method: 'POST',
+        url: 'http://localhost:3000/login',
+        data: {
+            email,
+            password
+        }            
+    })
+        .done(({data}) => {
+            console.log(data)
+            localStorage.token = data.token
+            localStorage.currentUser = data.name
+            listAll()
+            $('#loginForm').modal('hide')
+            $('.error-message').empty()
+            $('#btn-login').hide()
+            $('#btn-register').hide();
+            $('#dropdown-action').show()
+            $('#username-actions').empty()
+            $('#username-actions').append(`<img src="https://robohash.org/${data.name}?set=set5" style="width: 10%;" alt="assets/img/favicon/aeab52dcc9ee9b028f28623a7778de3b.ico"> ${data.name}`)
+        })
+        .fail(err => {
+            $('.error-message').empty()
+            $('.error-message').append(`<small class="form-text text-muted">${err.responseJSON.msg}</small>`)
+        })
+}
+
+function onSignIn(googleUser) {
+    const idToken = googleUser.getAuthResponse().id_token;
+    $.ajax({
+        method: 'POST',
+        url: 'http://localhost:3000/gLogin',
+        headers: {
+          idToken
```

Comment:
google sign in gagal gara2 ketika req.headers diterima key nya jadi kecil semua , jadi idtoken di server, coba ya